packages/cli/src/config/setup/Setup.handler.ts

```diff
@@ -87,10 +94,16 @@ export default class ServerInstallHandler implements ICommandHandler {
             Config.set(property.argLoc.jsonLoc, property.argValue, { secure: isSecured && property.secure });
         }
 
-        // biome-ignore lint/suspicious/noExplicitAny: Required `as any` to set profile type
-        (Config as any).mLayers.find(
-            (profile: { path: string }) => profile.path === newProperties[0].argLoc?.osLoc[0],
-        ).properties.profiles[profile.name].type = "ssh";
+        if (newProperties.length > 0 && newProperties[0].argLoc?.osLoc?.[0]) {
+            // biome-ignore lint/suspicious/noExplicitAny: Required `as any` to set profile type
+            const layer = (Config as any).mLayers.find(
```

Comment:
I know this wasn't added in this PR, but we should figure out a way to not have to do `as any` here. 
This will be something we need to fix/do upstream (in CLI)

Reply:
I'm not sure we need to make an upstream change for this, we can use [`Config.paths`](https://github.com/zowe/zowe-cli/blob/ee25ca00f0af31e5a763d88ef504bf0a36ead68e/packages/imperative/src/config/src/Config.ts#L302) here 😋

Reply:
We are already using `Config.set` to apply new fields on one of the profiles. Couldn't we do the same to set the type?

```suggestion
            Config.set(`profiles.${profile.name}.type`, "ssh");
```

Then we don't have to do the work to find what layer is affected by this change, and just let the Config API handle it for us 😋